### PR TITLE
Adding Success/Failure metric method

### PR DESCRIPTION
**Pull request by mgc1194**
Adding a method in the i18n/metrics module that logs into CloudWatch a success or failure message.
This PR introduces the method `report_success` that logs to CloudWatch the Success Status of a sync process.
It takes two required parameters:
- A true or false value that represents whether the process has been successful or not.
- A string representing the step of the sync where the method is used. (sync-in, sync-up, sync-down, sync-out)

And two optional parameters:
- A string representing the specific sync component being logged.
- A string containing the Exception message causing the process to fail.

## Links

- jira ticket: [P20-379](https://codedotorg.atlassian.net/browse/P20-379)

## Testing story
This PR adds unit test coverage for the method.
Additionally, I have done a test run and attached a screenshot of how this metric looks in CloudWatch
Screenshot 2023-09-07 at 10 16 40

## Follow-up work
Following this PR, the metrics methods need to be added to each sync module.

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: bin/i18n/metrics.rb [hunk 1]
```diff
@@ -38,6 +39,20 @@ def self.report_filesize(file_path, sync_step)
       )
     end
 
+    # logging to CloudWatch the Completion Status of a sync step, either success or fail.
+    # @param status [Boolean] Whether a step has been successful or not.
+    # @param sync_step [String] Step of the sync where the method is used. Options: in, up, down, out.
+    # @option sync_component [String] Specific sync component being logged.
+    # @option message [String] Exception message causing the step to fail.
+    def self.report_success(status, sync_step, sync_component = '', message = '')
```
Comment (annaxuphoto) [inline bin/i18n/metrics.rb]: Can you add a unit test for this method?

**Review by annaxuphoto — CHANGES_REQUESTED** [review]

File: bin/i18n/metrics.rb (hunk 1 above)
Comment (mgc1194) [reply bin/i18n/metrics.rb]: Absolutely!!

File: bin/i18n/metrics.rb [hunk 2]
```diff
@@ -39,15 +39,15 @@ def self.report_filesize(file_path, sync_step)
...
-    # logging to CloudWatch the Completion Status of a sync step, either success or fail.
+    # logging to CloudWatch the Completion Status of a sync process, either success or fail.
     # @param status [Boolean] Whether a step has been successful or not.
-    # @param sync_step [String] Step of the sync where the method is used. Options: in, up, down, out.
+    # @param sync_step [String] Step of the sync where the method is used. Options: sync-in, sync-up, sync-down, sync-out.
     # @option sync_component [String] Specific sync component being logged.
-    # @option message [String] Exception message causing the step to fail.
-    def self.report_success(status, sync_step, sync_component = '', message = '')
+    # @option message [String] Exception message causing the process to fail.
+    def self.report_success(status, sync_step, sync_component = 'None', message = 'None')
```
Comment (annaxuphoto) [inline bin/i18n/metrics.rb]: You can switch the order of the last two arguments if you don't want to specify `sync_component` when you're logging the `message` dimension.

**Review by annaxuphoto — APPROVED** [review]

**Review by artem-vavilov — APPROVED** [review]
I would split the method into two methods `.report_sync_success` and `.report_sync_failure` or rename it to `.report_sync_status(success, ...)`, but in general everything is great. Nice job!

File: bin/i18n/metrics.rb (hunk 2 above)
Comment (mgc1194) [reply bin/i18n/metrics.rb]: Made this change.

**Pull request merged**